### PR TITLE
Add pb116

### PR DESCRIPTION
**Pull request by talzich**
## Description
Add new validation PB116

### CODE REVIEW

File: .changelog/4364.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Add PB116 validation, playbook stops on error
```
Comment (YuvHayun) [inline .changelog/4364.yml:2]:
```suggestion
- description: Moved PB116 validation to the new format. the validation <what it does?>.
```

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ContentTypes = Playbook
+
+
+def is_indicator_pb(playbook: Playbook):
+    return any(
+        (i.get("playbookInputQuery") or {}).get("queryEntity") == "indicators" for i in playbook.data.get("inputs", {})
+    )
+
+
+def does_playbook_continue_on_error(playbook: Playbook):
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py:21]: Add doc strings for both functions.

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 3]
```diff
@@ -222,3 +226,72 @@ def test_IsAskConditionHasUnhandledReplyOptionsValidator():
...
+                "taskid": "27b9c747-b883-4878-8b60-7f352098a631",
+                "type": "condition",
+                "message": {"replyOptions": ["yes"]},
+                "nexttasks": {"no": ["1"]},
+                "continueonerror": True,
+                "task": {"id": "27b9c747-b883-4878-8b60-7f352098a63c"},
+
+            }
+        )
+    }
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:295]: Why not adding this to create_playbook_object.

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 4]
```diff
@@ -222,3 +226,72 @@ def test_IsAskConditionHasUnhandledReplyOptionsValidator():
...
+                "nexttasks": {"no": ["1"]},
+                "task": {"id": "27b9c747-b883-4878-8b60-7f352098a63c"},
+            }
+        )
+    }
+    res = IsStoppingOnErrorValidator().is_valid([playbook])
+    assert len(res) == 0
+
+
+def test_indicator_pb_must_stop_on_error_invalid():
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:268]: add given-when-then

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Nice!
Minor changes.

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py (hunk 2 above)
Comment (talzich) [reply demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py]: Done

File: demisto_sdk/commands/validate/tests/PB_validators_test.py (hunk 3 above)
Comment (talzich) [reply demisto_sdk/commands/validate/tests/PB_validators_test.py]: Done

File: demisto_sdk/commands/validate/tests/PB_validators_test.py (hunk 4 above)
Comment (talzich) [reply demisto_sdk/commands/validate/tests/PB_validators_test.py:262]: Done

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+from demisto_sdk.commands.content_graph.objects.playbook import Playbook
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Playbook
+
+
+def is_indicator_pb(playbook: Playbook):
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py:15]: @aaron1535 refactor to use this function
@YuvHayun is there a better place to put this? we have 3 validations that should use it

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    Retrieve tasks from the playbook that are set to continue on error.
+    """
+    return [task for task in playbook.tasks.values() if task.continueonerror]
+
+
+class IsStoppingOnErrorValidator(BaseValidator[ContentTypes]):
+    error_code = "PB116"
+    description = "The validation checks that all playbook tasks stop when encountering an error."
+    rationale = "If a playbook task does not stop on error, following tasks might rely on its output and fail."
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py:34]: we should explain why this is this only for indicator pbs. lets also write in the description that its specific to this type

**Review by JudahSchwartz — APPROVED** [review]
fantastic, note my comments

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py (hunk 5 above)
Comment (talzich) [reply demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py]: Done

File: demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py (hunk 6 above)
Comment (talzich) [reply demisto_sdk/commands/validate/validators/PB_validators/PB116_is_stopping_on_error.py]: done

**Pull request merged**